qipr_approver/approver/decorators/login_required.py

```diff
@@ -10,7 +10,11 @@ def login_required(view_function):
     """
     def wrapped_view(request, *args, **kwargs):
         if request.session.get(SESSION_VARS.get('gatorlink')):
-            return view_function(request, *args, **kwargs)
-        else:
-            return redirect(reverse("approver:shib"))
+            if __shib_validated(request):
+                return view_function(request, *args, **kwargs)
+        return redirect(reverse("approver:shib"))
+
+    def __shib_validated(request):
+        return ((SHIB_ENABLED == 'true' and request.META.get('HTTP_EPPN')) or (SHIB_ENABLED == 'false'))
```

Comment:
I'll leave a comment, forgot to... Essentially, if shib is working and authenticated, it creates/populates this HTTP_EPPN meta-data. We only care if there is something in EPPN if shib is actually turned on. Otherwise, we don't care. I don't want it to get stuck in a state where the session expired (no gatorlink), but still has HTTP_EPPN, or vice versa